shuup/admin/modules/media/views.py

```diff
@@ -64,18 +69,69 @@ def _filer_folder_to_json_dict(folder, children=None):
     }
 
 
+def _ensure_media_folder(shop, folder):
+    media_folder, created = MediaFolder.objects.get_or_create(folder=folder)
+    if not media_folder.shops.filter(id=shop.id).exists():
+        media_folder.shops.add(shop)
+
+
+def _is_folder_shared(folder):
+    if not settings.SHUUP_ENABLE_MULTIPLE_SHOPS:
+        return False
+
+    media_folder = MediaFolder.objects.filter(folder=folder).first()
+    if not media_folder:
+        return True
+    return bool(media_folder.shops.count() > 1)
```

Comment:
hum.. the folder is shared only when `count() > 1`? what if `shops.count() == 0` ?
also, should shop3 has access to the file which is set for shop1 and shop2? at least this will be a valid test in this case